### PR TITLE
[SPARK-34224][CORE][SQL][SS][DSTREAM][YARN][TEST][EXAMPLES][2.4] Ensure all resource opened by `Source.fromXXX` are closed 

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
Using a function like `.mkString` or `.getLines` directly on a `scala.io.Source` opened by `fromFile`, `fromURL`, `fromURI ` will leak the underlying file handle,  this pr use the `Utils.tryWithResource` method wrap the `BufferedSource` to ensure these `BufferedSource` closed.

### Why are the changes needed?
Avoid file handle leak.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Pass the Jenkins or GitHub Action

### CODE REVIEW

File: core/src/test/scala/org/apache/spark/deploy/master/MasterSuite.scala [hunk 1]
```diff
@@ -264,16 +265,17 @@ class MasterSuite extends SparkFunSuite
     val conf = new SparkConf()
     val localCluster = new LocalSparkCluster(2, 2, 512, conf)
     localCluster.start()
+    val masterUrl = s"http://localhost:${localCluster.masterWebUIPort}"
```
Comment (LuciferYang) [inline core/src/test/scala/org/apache/spark/deploy/master/MasterSuite.scala:268]: Similar to https://github.com/apache/spark/pull/31344, Not easy to maintain the same format as the master, so added masterUrl(line 268)

Comment (LuciferYang) [inline core/src/test/scala/org/apache/spark/deploy/master/MasterSuite.scala:268]: Similar to https://github.com/apache/spark/pull/31344, Not easy to maintain the same format as the master, so added masterUrl(line 268)

**Review by HyukjinKwon — APPROVED** [review]

File: core/src/test/scala/org/apache/spark/deploy/master/MasterSuite.scala (hunk 1 above)
Comment (HyukjinKwon) [reply core/src/test/scala/org/apache/spark/deploy/master/MasterSuite.scala:268]: It's fine. thanks for the backport PRs

**Comment by HyukjinKwon** [comment]
@LuciferYang can you fix the build please?

**Comment by LuciferYang** [comment]
@HyukjinKwon I'm trying, the problem is `BufferedSource` is not extends `java.io.Closeable` , so the result of  `Source.fromXXX` cannot be an argument to the `Utils. tryWithResource ` method

**Comment by LuciferYang** [comment]
@HyukjinKwon Is there any other suggestion besides using the  `try {} finally {}` instead?

**Comment by HyukjinKwon** [comment]
Ah, I see .. let's just drop this then. I think it's not worthwhile if we have to introduce some diff.

**Comment by LuciferYang** [comment]
OK, I will close this pr, thx ~ @HyukjinKwon

**Comment by LuciferYang** [comment]
If I find a low-cost way, I will reopen this pr

**Comment by HyukjinKwon** [comment]
Thank you @LuciferYang :-)

**Pull request closed without merging**